vars/dockerBuildArgs.groovy

```diff
@@ -48,27 +100,46 @@ String call(Map config = [:]) {
       ret_str += " --build-arg CACHEBUST=${currentBuild.startTimeInMillis}"
       ret_str += " --build-arg CB0=" + current_time.get(Calendar.WEEK_OF_YEAR)
     }
-
     if (add_repos) {
-      if (env.REPOSITORY_URL) {
-        ret_str += ' --build-arg REPO_URL=' + env.REPOSITORY_URL
-      }
-      if (env.DAOS_STACK_EL_7_LOCAL_REPO) {
-        ret_str += ' --build-arg REPO_EL7=' + env.DAOS_STACK_EL_7_LOCAL_REPO
+      String repo_name = null
+      if (stage_info['target'] == 'centos7') {
+        repo_alias = 'EL_7'
+        repo_name = env["DAOS_STACK_${repo_alias}${repo_mod}_${repo_type}_REPO"]
+        if (repo_name) {
+          ret_str += ' --build-arg REPO_EL7=' + repo_name
+        }
       }
-      if (env.DAOS_STACK_EL_8_LOCAL_REPO) {
-        ret_str += ' --build-arg REPO_EL8=' + env.DAOS_STACK_EL_8_LOCAL_REPO
+      if (stage_info['target'] == 'centos8') {
+        repo_alias = 'EL_8'
+        repo_name = env["DAOS_STACK_${repo_alias}${repo_mod}_${repo_type}_REPO"]
+        if (repo_name) {
+          ret_str += ' --build-arg REPO_EL8=' + repo_name
+        }
       }
-      if (env.DAOS_STACK_LEAP_15_LOCAL_REPO) {
-        ret_str += ' --build-arg REPO_LOCAL_LEAP15=' +
-                   env.DAOS_STACK_LEAP_15_LOCAL_REPO
+      if (stage_info['target'] == 'leap15') {
+        repo_alias = 'LEAP_15'
+        repo_name = env["DAOS_STACK_${repo_alias}${repo_mod}_${repo_type}_REPO"]
+        if (repo_name) {
+          ret_str += ' --build-arg REPO_EL7=' + repo_name
+        }
+        if (repo_type == 'LOCAL') {
+         if (env.DAOS_STACK_LEAP_15_GROUP_REPO) {
+            ret_str += ' --build-arg REPO_GROUP_LEAP15=' +
+                       env.DAOS_STACK_LEAP_15_GROUP_REPO
+          }
+        }
       }
-      if (env.DAOS_STACK_UBUNTU_20_04_LOCAL_REPO) {
-        ret_str += ' --build-arg REPO_UBUNTU_20_04=' + env.DAOS_STACK_UBUNTU_20_04_LOCAL_REPO
+      if (stage_info['target'] == 'ubuntu20.04') {
+        repo_name = env.DAOS_STACK_UBUNTU_20_04_LOCAL_REPO
+        if (repo_type == 'LOCAL') {
+          repo_name = env.DAOS_STACK_UBUNTU_20_04_LOCAL_REPO
+          if (repo_name) {
+            ret_str += ' --build-arg REPO_UBUNTU_20_04=' + repo_name
+          }
+        }
```

Comment:
This will get folded into the above more concise processing at some point?

Reply:
Will be done in the next push.   We are not using the UBUNTU local repo now, and we have not setup the infrastructure for proxying the other Ubuntu repositories yet.